### PR TITLE
Initial

### PR DESCRIPTION
**Pull request by armw4**
see [README.md](https://github.com/lanetix/elasticsearch-filter-append/tree/initial)

### CODE REVIEW

**Comment by armw4** [comment]
@lanetix/engineers

File: package.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  "homepage": "https://github.com/lanetix/elasticsearch-filter-append#readme",
+  "devDependencies": {
+    "mocha": "^2.3.3",
+    "standard": "^5.3.1"
+  },
+  "dependencies": {
+    "lodash.clone": "^3.0.3",
+    "lodash.isarray": "^3.0.4",
+    "lodash.toarray": "^3.0.2",
+    "should": "^7.1.1"
```
Comment (apechimp) [inline package.json]: Should be devDependency

Comment (armw4) [inline package.json]: That's right. lemme do that and doc constructor.

**Comment by armw4** [comment]
@apechimp anymore feedback for me, sir?

File: lib/append.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      ]
+    }
+  }
+}
+
+module.exports = function (parent, filter, filters) {
+  if (filter.bool) {
+    if (filter.bool.must) {
+      appendToMustFilter(filter, filters)
+    } else if (filter.bool.hasOwnProperty('should')) {
```
Comment (armw4) [inline lib/append.js]: this is because of `should.js` monkey patching all things `Object`.

Comment (apechimp) [inline lib/append.js]: oh yeah...that's annoying. Consider just using `assert` :wink:

**Comment by apechimp** [comment]
Belated :shipit:

File: lib/append.js (hunk 2 above)
Comment (armw4) [inline lib/append.js]: Indeed.

**Pull request merged**